src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -192,6 +514,36 @@ protected SnapshotContext<YBPartition, YugabyteDBOffsetContext> prepare(YBPartit
         return new YugabyteDBSnapshotContext(partition, connectorConfig.databaseName());
     }
 
+    /**
+     * Check on the server side if the tablet already has some checkpoint, if it does then do
+     * not take a snapshot for it again since some data has already been streamed out of it
+     * @param tableId the UUID of the table
+     * @param tabletId the UUID of the tablet
+     * @return true if snapshot has been taken or some data has streamed already
+     * @throws Exception if checkpoint cannot be retreived from server side
+     */
+    protected boolean hasSnapshotCompletedPreviously(String tableId, String tabletId) 
+        throws Exception {
+      // The default behaviour is configured to allow the connector to take a snapshot again
+      // and returning false at this stage would indicate that the snapshot has not been taken
+      // so the connector would try to take a snapshot considering this as the first time
+      // it is going for the snapshot
+      if (this.connectorConfig.snapshotAgain()) {
```

Comment:
Note to self: handle scenario where all the tablets are in streaming mode and the connector is restarted.